### PR TITLE
docs: add TypeScript types declaration file

### LINKED ISSUE #62: TypeScript type declaration file
If you want I can write a TypeScript type declaration file for this package so that TS developers can utilise the type-checking feature of the language as they use this package. Let me know if you are interested.

### PR DESCRIPTION
**Pull request by geopic**
Resolves #62. Let me know if there are any issues.

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+     *   'creation date': 1303979726,
+     *   info: { length: 718583808,
+     *     name: 'ubuntu-11.04-desktop-i386.iso',
+     *     'piece length': 524288,
+     *     pieces: <Buffer e5 7a ...>
+     *   }
+     * }
+     * ```
+     */
+    metadata: object;
```
Comment (fent) [inline lib/index.d.ts:55]: do you mind adding the details of the metadata object? looking at `schema.js` it looks something like

```js
{
  announce: string;
  'announce-list'?: string[][]
  'creation-date'?: string;
  comment?: string;
  'created-by'?: string;
  encoding?: string;
  info: MultiFileInfo | SingleFileInfo;
};

type Info = {
    name?: string;
    'piece-length': int;
    pieces: string;
    private?: 0 | 1;
    source?: string;
  };
type MultiFileInfo = Info & {
  files: {
    length: int;
    path: string;
    md5sum?: string;
  }[];
}
type SingleFileInfo = Info & {
  // maybe it needs `name` too?
  length: int;
  md5sum?: string;
}
```

**Review by fent — CHANGES_REQUESTED** [review]
thank you! very nicely done

added a couple of comments

File: lib/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+  namespace NodeTorrent {
+    /**
+     * Reads a local file, or a readable stream. Returns readable stream.
+     * 
+     * An error can be returned if the torrent is formatted incorrectly.
+     * Does not check if the dictonaries are listed alphabetically.
+     * Refer to the [BitTorrent Specification](https://wiki.theory.org/BitTorrentSpecification) for more info on torrent metainfo.
+     *
+     * @param file Where the torrent resides, can be local file, remote, or a readable stream.
```
[the remark this answers is not in the document]
Comment (geopic) [reply lib/index.d.ts:139]: Noted

File: lib/index.d.ts (hunk 1 above)
Comment (geopic) [reply lib/index.d.ts:55]: Thank you for typing it out for me. :+1: I think if `SingleFileInfo` includes the `Info` type then it won't need `name` specified

**Review by fent — APPROVED** [review]
thank you!

**Pull request merged**